# Pipeline_dual2019_miseq- (ID: MYPBR2019CDv1)

Bioinformatics pipeline to process fastq miseq for target sequencing : dual calling of variants



![pipeline](https://github.com/cdesterke/Pipeline_dual2019_miseq-/blob/master/pipeline.png)

# references

    - Lai Z, Markovets A, Ahdesmaki M, Chapman B, Hofmann O, McEwen R, Johnson J,Dougherty B, Barrett JC, Dry JR. VarDict: a novel and versatile variant caller for next-generation sequencing in cancer research. Nucleic Acids Res. 2016 Jun 20;44(11):e108.
    - Langmead B, Salzberg SL. Fast gapped-read alignment with Bowtie 2. Nat Methods. 4 mars 2012;9(4):357‑9. 
    - Li H, Handsaker B, Wysoker A, Fennell T, Ruan J, Homer N, et al. The Sequence Alignment/Map format and SAMtools. Bioinforma Oxf Engl. 15 août 2009;25(16):2078‑9. 
    - Quinlan AR. BEDTools: The Swiss-Army Tool for Genome Feature Analysis. Curr Protoc Bioinforma. 8 sept 2014;47:11.12.1-34. 
    - Van der Auwera GA, Carneiro MO, Hartl C, Poplin R, Del Angel G, Levy-Moonshine A, et al. From FastQ data to high confidence variant calls: the Genome Analysis Toolkit best practices pipeline. Curr Protoc Bioinforma. 2013;43:11.10.1-33. 
    - Wang K, Li M, Hakonarson H. ANNOVAR: functional annotation of genetic variants from high-throughput sequencing data. Nucleic Acids Res. sept 2010;38(16):e164. 
